### DISCUSSION TITLE
[Projects Beta] Custom Project Templates

### DISCUSSION THREAD

**Discussion by odyvangelis** [community/community#8023] [Projects and Issues] [answered]
I would like to use the new Projects to plan sprints, essentially creating a new Project per sprint. 

For this to be actually useful, there needs to be a way to either clone/copy an existing project or define a template that you can use when creating a new one, with predefined views and fields. 

I believe this would be valuable for other use cases as well.

**Comment by joshuaNathaniel** [comment, marked as answer]
I suggest you add an iteration field called _Sprint_ and just filter a board view by the sprint. Make your life easier.

**Reply by odyvangelis** [reply]
I guess that makes the most sense with the current implementation. Thanks for the suggestion!

**Reply by manusa** [reply]
I'm not sure how you add the "iteration field", haven't seen the option to add new fields yet.

However, if I want to keep a copy of the current Sprint Board for future reference (which is what we do now). After a few Sprints, lets say 10, the tab list would become cluttered. It would also be very hard to find a tab from 20-30 iterations ago.

I agree that this is the most "relational" way of doing things, but the current UI doesn't seem designed for this approach. If this is the officially accepted workaround, I'd suggest to improve the previously highlighted points.

**Reply by odyvangelis** [reply]
@manusa on one of your project's Table Views, if you scroll to the right and click the '+' icon, you will see a 'New Field' option:
![image](https://user-images.githubusercontent.com/39122638/149221706-88898054-cb08-4328-9faa-4d13e506acfd.png)

You can then add an 'Iteration' field and name it Sprint. You can set the default sprint duration when creating the Iteration field. 
Then, you can create a new view and filter it by Sprint. There now exists the `@current` iteration filter, so you can create something like this:
![image](https://user-images.githubusercontent.com/39122638/149221978-4692fd75-5837-4edb-978d-66fb387ab520.png)

If you want to see a specific sprint you can substitute the Current keyword for the specific sprint. No need for a new tab for each one. 

I think that when more advanced Workflows are released, it will be a good enough solution for small teams.

**Reply by manusa** [reply]
Cool, we'll definitely try this out.
Thanks for your detailed explanation.

**Comment by baden-nira** [comment]
The "non-sprint" element of the original request, ie 

"there needs to be a way to either clone/copy an existing project (beta) or define a template that you can use when creating a new one, with predefined views and fields."

is something we'd really find useful.

**Comment by vilbergs** [comment]
I am evaluating whether the new GitHub projects feature could work well with the Shape Up methodology. A part of Shape Up is that a team takes full ownership over a project, and so I want to be able to have a GitHub project for each Shape Up project, so that access can be controlled per project team.

Project templates / cloning seems essential for this to be viable for us.
